KarinaKlevzhits/hometask_4/spec/lib/my_hash_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require_relative '../../my_hash'
+require 'rspec'
+
+RSpec.describe MyHash do
+  subject(:my_hash) { described_class.new }
+
+  describe '#[]=' do
+    before { my_hash[:a] = 5 }
+
+    context 'when key does not exist' do
+      it { expect(my_hash[:a]).to eq(5) }
+    end
+
+    context 'when not valid parametrs of key' do
+      it { expect { my_hash[] = 5 }.to raise_error(ArgumentError) }
+    end
+  end
+
+  describe '#[]' do
+    before { my_hash[:key] = 'hello' }
+
+    context 'when key exists' do
+      it { expect(my_hash[:key]).to eq('hello') }
+    end
+
+    context 'when key does not exists' do
+      it { expect(my_hash[:key2]).to eq(nil) }
+    end
+  end
+
+  describe '#delete' do
+    before { my_hash[:key] = 'ruby' }
+
+    context 'when key exists' do
+      before { my_hash.delete(:key) }
+
+      it { expect(my_hash[:key]).to eq(nil) }
+    end
+
+    context 'when key does not exists' do
+      it { expect(my_hash.delete(:key2)).to eq(nil) }
+    end
+
+    context 'when wrong parametrs in method delete' do
+      it { expect { my_hash.delete }.to raise_error(ArgumentError) }
+    end
+  end
+
+  describe '#clear' do
```

Comment:
If I try to cover this method I'll add 3 test:
1. cover result of clear hash( expect(my_hash.clear).to eq({}) );
2. cover what my hash return after clear( expect(my_hash).to eq({}) ); 
3. cover length of hash after clear(you've done this, but rename 'it' to something like 'return length of my hash equal 0');
4. cover that key-value is unavailable after clear(that also you realized, and try to rename to 'does not return value for key 2').
Change contexts to 'when my hash cleared'(2,3,4) and 'when my hash is clearing'(1)